### PR TITLE
In the scope of backgroundables get the project from GitRepository instead of constructor argument

### PR DESCRIPTION
**Pull request by mkondratek**

### CODE REVIEW

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutAndDeleteHandler.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.actions.dialogs.DeleteBranchOnSlideOutSuggestionDialog;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseEnhancedGraphTable;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod(GitMacheteBundle.class)
+@CustomLog
+public class SlideOutAndDeleteHandler {
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutAndDeleteHandler.java:32]: the main reason for extracting this class is that this is a scope where `gitRepository` is not null

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutBackgroundable.java [hunk 2]
```diff
@@ -40,32 +23,20 @@ public class SlideOutBackgroundable extends Task.Backgroundable {
...
   private final BranchLayout branchLayout;
-
-  private final IBranchLayoutWriter branchLayoutWriter;
-
-  @Nullable
-  private final GitRepository selectedGitRepository;
-
   @Nullable
+  private final GitRepository gitRepository;
   private final BaseEnhancedGraphTable graphTable;
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutBackgroundable.java:28]: it was not nullable for some time already!

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutAndDeleteHandler.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+          getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-success.without-delete.HTML")
+              .fmt(branchName));
+    }
+    if (graphTable != null) {
+      graphTable.queueRepositoryUpdateAndModelRefresh();
+    }
+  }
+
+  @UIThreadUnsafe
+  private void slideOutBranch(String branchName) {
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutAndDeleteHandler.java:133]: @MaciejG604, pls confirm I did not reproduce https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1333 here XD

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutBackgroundable.java [hunk 4]
```diff
@@ -40,172 +23,39 @@ public class SlideOutBackgroundable extends Task.Backgroundable {
...
-  private void handleBranchDeletionDecision(String branchName, GitRepository gitRepository, boolean shouldDelete) {
-    slideOutBranch(branchName);
-    if (shouldDelete) {
-      GitBrancher.getInstance(project).deleteBranch(branchName, Collections.singletonList(gitRepository));
-      VcsNotifier.getInstance(project).notifySuccess(/* displayId */ null,
-          /* title */ "",
-          getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-success.with-delete.HTML")
-              .fmt(branchName));
-      return;
+    if (gitRepository != null) {
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutBackgroundable.java:49]: I am thinking... does it even make sense to distinguish the case when gitRepository is null? the run does nothing it such case 🤔 

maybe let's make gitRepository not-null in this scope already, and bring the SlideOutAndDelete handler back here? wdyt @PawelLipski ?

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/FetchBackgroundable.java [hunk 5]
```diff
@@ -28,15 +26,14 @@ public class FetchBackgroundable extends Task.Backgroundable {
...
+  public FetchBackgroundable(
       GitRepository gitRepository,
       String remoteName,
       String refspec,
       String taskTitle,
       @Untainted String failureNotificationText,
       String successNotificationText) {
-    super(project, taskTitle, /* canBeCancelled */ true);
-    this.project = project;
+    super(gitRepository.getProject(), taskTitle, /* canBeCancelled */ true);
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/FetchBackgroundable.java:36]: A high-level sanity check... obviously the result of `getProject()` is fixed/immutable for the given instance of `GitRepository` right? 😅

Comment (mkondratek) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/FetchBackgroundable.java:36]: yup,
the method `getProject` is `@NotNull` in the `Repository` interface, in the implementation `RepositoryImpl` and as a field:
image

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutAndDeleteHandler.java (hunk 3 above)
Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutAndDeleteHandler.java:133]: If there's no `assert ... != null`, then we're rather safe ;)

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutAndDeleteHandler.java (hunk 1 above)
Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutAndDeleteHandler.java:38]: I'm assuming there's :zero: changes in logic here 😅

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutBackgroundable.java (hunk 4 above)
Comment (PawelLipski) [reply frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/SlideOutBackgroundable.java:49]: +1, I'd go for passing `gitRepository` to this class as `@NonNull` already

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectDependentAction.java [hunk 6]
```diff
@@ -62,10 +60,6 @@ protected Project getProject(AnActionEvent anActionEvent) {
     return project;
   }
 
-  protected IBranchLayoutWriter getBranchLayoutWriter() {
-    return RuntimeBinding.instantiateSoleImplementingClass(IBranchLayoutWriter.class);
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseProjectDependentAction.java:66]: +1, no point in providing this one here

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**